lib/bloc/user_profile/user_profile_bloc.dart

```diff
@@ -26,23 +26,18 @@ class UserProfileBloc extends Cubit<UserProfileState> with HydratedMixin {
     var profile = state;
     _log.v("State: ${profile.profileSettings.toJson()}");
     final settings = profile.profileSettings;
-    if (settings.color == null || settings.animal == null) {
+    if (settings.color == null && settings.animal == null) {
```

Comment:
It needs to be an `or` not an `and`. If you are adding the name to the same if statement you need to add the `name == null` as a third or clause.

Reply:
Added correct checking as well as generating the name properly.